elasticsearch/client.go

```diff
@@ -18,81 +18,88 @@
 package elasticsearch
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	"errors"
 	"io"
+	"io/ioutil"
 	"net/http"
+	"net/url"
+	"strings"
 
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/version"
 
 	v7 "github.com/elastic/go-elasticsearch/v7"
-	v7esapi "github.com/elastic/go-elasticsearch/v7/esapi"
 
 	v8 "github.com/elastic/go-elasticsearch/v8"
-	v8esapi "github.com/elastic/go-elasticsearch/v8/esapi"
 )
 
 // Client is an interface designed to abstract away version differences between elasticsearch clients
 type Client interface {
+	// TODO: deprecate
```

Comment:
the line bellow it, the `Search` interface. Mostly because it is redundant now:
it can be implemented with the `JsonRequest` and richer data types (as functions in `security_api.go`, instead of requiring and returning low level readers.

It would also resolve colliding names, which would allow us to embed the whole client struct in the wrapper, avoiding the need for more (confusing) names: https://github.com/elastic/apm-server/pull/3063#discussion_r359246369

Reply:
Are you planning on doing this within this PR? If not, please create an issue with a small description so someone can pick it up and maybe link it here to the TODO.